packages/odyssey-react-mui/src/labs/DataFilters.tsx

```diff
@@ -367,23 +391,53 @@ const DataFilters = ({
                     {/* Autocomplete */}
                     {filterPopoverCurrentFilter?.variant === "autocomplete" &&
                       filterPopoverCurrentFilter?.options && (
-                        <Autocomplete
-                          label={filterPopoverCurrentFilter.label}
-                          value={
-                            (inputValues[
-                              filterPopoverCurrentFilter.id
-                            ] as string) ?? ""
-                          }
-                          onBlur={function ro() {}}
-                          onChange={function ro() {}}
-                          onFocus={function ro() {}}
-                          onInputChange={function ro() {}}
-                          options={filterPopoverCurrentFilter.options.map(
-                            (option: { label: string }) => ({
-                              label: option.label,
-                            })
-                          )}
-                        />
+                        <Box
+                          sx={{
+                            display: "flex",
+                            gap: 2,
+                            alignItems: "flex-end",
+                          }}
+                        >
+                          <Box sx={{ width: "100%" }}>
+                            <Autocomplete
+                              label={filterPopoverCurrentFilter.label}
+                              value={
+                                inputValues[filterPopoverCurrentFilter.id] ?? ""
+                              }
+                              onChange={(_, value) => {
+                                const label =
+                                  typeof value === "string"
+                                    ? getAutoCompleteLabel({ label: value })
+                                    : Array.isArray(value)
+                                    ? getAutoCompleteLabel(
+                                        value.map((item) =>
+                                          typeof item === "string"
+                                            ? { label: item }
+                                            : item
+                                        )
+                                      )
+                                    : value
+                                    ? getAutoCompleteLabel(value)
+                                    : "";
+
+                                updateInputValue({
+                                  filterId: filterPopoverCurrentFilter.id,
+                                  value: label,
+                                });
+                              }}
+                              options={filterPopoverCurrentFilter.options.map(
+                                (option: { label: string }) => ({
+                                  label: option.label,
+                                })
+                              )}
```

Comment:
We need to memoize this value.

Right now, we're creating `options` every time a re-render occurs which is anytime you click something in `DataFilters`, but I don't think the options ever change once this is rendered unless we're removing selected filters?

I thought there was something @jordankoschei-okta did recently about making it so selected items stay in the `Autocomplete` list.

Reply:
@KevinGhadyani-Okta Are you open to pairing on this? It looks like we use `memo` elsewhere in this file but I am not familiar with how the library works.